src/windows/common/PortableDistribution.cpp

```diff
@@ -0,0 +1,435 @@
+        
+        if (!distroKey)
+        {
+            return std::nullopt;
+        }
+
+        wil::unique_cotaskmem_string pathStr;
+        if (SUCCEEDED(wsl::windows::common::registry::ReadString(distroKey.get(), nullptr, c_portableRegistryValue, pathStr)))
+        {
+            return std::filesystem::path(pathStr.get());
+        }
+    }
+    catch (...) {}
+
+    return std::nullopt;
+}
+
+void ValidatePortablePath(_In_ const std::filesystem::path& path, _In_ bool allowFixed)
+{
+    // Check if path exists
+    THROW_HR_IF_MSG(
+        HRESULT_FROM_WIN32(ERROR_PATH_NOT_FOUND),
+        !std::filesystem::exists(path),
+        "Portable path does not exist: %ls",
+        path.c_str());
+
+    // Check if it's a directory
+    THROW_HR_IF_MSG(
+        HRESULT_FROM_WIN32(ERROR_DIRECTORY),
+        !std::filesystem::is_directory(path),
+        "Portable path must be a directory: %ls",
+        path.c_str());
+
+    // Check if it's on removable media (restrict to removable by default)
+    THROW_HR_IF_MSG(
+        HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED),
+        !IsRemovableDrive(path, allowFixed),
+        "Portable path must be on removable media: %ls. To allow fixed drives, use the --allow-fixed flag.",
+        path.c_str());
+}
+
+void CreatePortableDistribution(
+    _In_ const std::filesystem::path& portablePath,
+    _In_ LPCWSTR distroName,
+    _In_ const std::filesystem::path& sourceFile,
+    _In_ ULONG version,
+    _In_ ULONG flags,
+    _In_ bool allowFixed)
+{
+    // Validate portable path
+    ValidatePortablePath(portablePath, allowFixed);
+
+    // Check if metadata already exists
+    auto metadataPath = portablePath / c_portableMetadataFileName;
+    THROW_HR_IF_MSG(
+        HRESULT_FROM_WIN32(ERROR_FILE_EXISTS),
+        std::filesystem::exists(metadataPath),
+        "Portable distribution already exists at: %ls",
+        portablePath.c_str());
+
+    // Determine target VHDX path
+    std::wstring vhdxFileName = distroName;
+    vhdxFileName += L".vhdx";
+    auto vhdxPath = portablePath / vhdxFileName;
+
+    // Import the distribution using WSL service
+    wsl::windows::common::SvcComm service;
+    
+    wil::unique_hfile sourceFileHandle;
+    sourceFileHandle.reset(CreateFileW(
+        sourceFile.c_str(),
+        GENERIC_READ,
+        FILE_SHARE_READ | FILE_SHARE_DELETE,
+        nullptr,
+        OPEN_EXISTING,
+        FILE_ATTRIBUTE_NORMAL,
+        nullptr));
+    
+    THROW_LAST_ERROR_IF(!sourceFileHandle);
+
+    // Create the VHDX in the portable location
+    auto [guid, name] = service.RegisterDistribution(
+        distroName,
+        version,
+        sourceFileHandle.get(),
+        portablePath.c_str(),
+        flags | LXSS_IMPORT_DISTRO_FLAGS_VHD);
+
+    // Unregister immediately as we only wanted to create the VHDX
+    try
+    {
+        service.UnregisterDistribution(&guid);
```

Comment:
Doesn't unregistering the distribution delete the VHDX as well on this code path ?